packages/components/src/DataList/components/DataListLayoutInternal/DataListItemInternal.tsx

```diff
@@ -0,0 +1,42 @@
+import classNames from "classnames";
+import React from "react";
+import { Checkbox } from "../../../Checkbox";
+import { useDataListContext } from "../../context/DataListContext";
+import styles from "../../DataList.css";
+import { DataListObject } from "../../DataList.types";
+
+interface ListItemInternalProps<T extends DataListObject> {
+  children: JSX.Element;
+  item: T;
+}
+export function ListItemInternal<T extends DataListObject>({
```

Comment:
mostly because file name and grouping

```suggestion
}

export function DataListItemInternal<T extends DataListObject>({
```